locale/en/docs/topics/async.md

```diff
@@ -0,0 +1,461 @@
+_after_ the called function has returned, there is no guarantee that the
+context is the same as when the original function was called. In this case the
+callback should be written in a way that makes no assumptions about the
+current context.
+
+One common way to address this is to save the context in a variable which is
+explicitly used in the callback. Another is to use the `bind()` function to
+create a function which calls the callback with the appropriate context.
+
+```js
+function MyObject() {
+  this.total = 0;
+}
+MyObject.prototype = {
+
+  increment: function() {
+    this.total += 1;
+    console.log('--TOTAL', this.total);
+  },
+
+  run: function() {
+    // TROUBLE! By the time the `increment()` function is called the `this`
+    // variable will no longer be its scope.
+    setTimeout(this.increment, 100);
+
+    // One approach is to save the context so that we can use it explicitly
+    // later.
+    var self = this;
+    setTimeout(function() {
+      self.increment();
+    }, 200);
+
+    // Another approach is to bind the function to the specific desired
+    // context.
+    setTimeout(this.increment.bind(this), 300);
+  }
+
+};
+var obj = new MyObject();
+obj.run();
+```
+
+
+### The Event Loop
+The event loop is a low-level part of node.js which keeps track of
+asynchronous operations which haven't completed.
+
+The event loop doesn't usually cause difficulty itself. However sometimes it's
+good to know about it because of how it impacts the order in which your code
+runs.
+
+Consider this example:
+
+```js
+console.log('--begin');
+setTimeout(function() {
+    console.log('--timeout');
+}, 100);
+console.log('--end');
+```
+This prints:
+
+```
+--begin
+--end
+--timeout
+```
+
+Even though the bottom of the script (the `--end`) is reached quickly, node.js
+knows that there is still a timeout pending and so doesn't let the script exit
+until that timeout has happened.
+
+Another thing which can keep the event loop from completing is if the process
+is listening for incoming network connections. This is how an HTTP server can
+keep running, waiting for page requests, even though the main program has
+reached the end.
+
+The "event loop" has that name because it is a "loop" which runs continuously,
+looking for "events" which have happened and processing them. It will continue
+to loop until it is no longer expecting more events to happen.
+
+The event loop works something like the following pseudo-code:
+
+```js
+while (true) {
+  for (each event that happened) {
+    process the event
+  }
+  if (no more pending events) {
+    break;
+  }
+}
+```
+
+Some examples of events for which the event loop will wait:
+
+* An asynchronous filesystem operation has completed.
+* An asynchronous network operation has completed.
+* A new incoming network connection was detected.
+* A timer has timed out.
+
+There are two functions which allow you to directly manipulate the event loop:
+`nextTick()` and `setImmediate()`.
```

Comment:
might be worth mentions on the "It is very important for APIs to be either 100% synchronous or 100%
asynchronous.  This is a important and misunderstood part of Node. Docs here https://nodejs.org/dist/latest-v4.x/docs/api/process.html#process_process_nexttick_callback_arg